extras/ai-battle/main.cpp

```diff
@@ -0,0 +1,124 @@
+// Copyright (C) 2005 - 2024 Settlers Freaks (sf-team at siedler25.org)
+//
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "GlobalGameSettings.h"
+#include "HeadlessGame.h"
+#include "QuickStartGame.h"
+#include "RTTR_Version.h"
+#include "RttrConfig.h"
+#include "files.h"
+#include "random/Random.h"
+#include "s25util/System.h"
+
+#include <boost/filesystem.hpp>
+#include <boost/nowide/args.hpp>
+#include <boost/nowide/filesystem.hpp>
+#include <boost/nowide/iostream.hpp>
+#include <boost/optional.hpp>
+#include <boost/program_options.hpp>
+
+namespace bnw = boost::nowide;
+namespace bfs = boost::filesystem;
+namespace po = boost::program_options;
+
+int main(int argc, char** argv)
+{
+    bnw::nowide_filesystem();
+    bnw::args _(argc, argv);
+
+    boost::optional<std::string> replay_path;
+    boost::optional<std::string> savegame_path;
+    unsigned random_init = static_cast<unsigned>(std::chrono::high_resolution_clock::now().time_since_epoch().count());
```

Comment:
```suggestion
    const unsigned random_init = static_cast<unsigned>(std::chrono::high_resolution_clock::now().time_since_epoch().count());
```

forgot const.

Reply:
done

Reply:
That was on purpose: The idea was to make this the default, i.e.: `po::value(&random_init)->default_value(defaultSeed)`
Or (I think) you can init  `random_init` here and use it as an implicit default. But in either case it must be non-const such that it can be set by the cmdline parser